### PR TITLE
Finished Creating Columns.java

### PR DESCRIPTION
**Pull request by loughrao**
Solves issues #45 I haven't removed the column based functions from game.java yet as we will need to change the constructor before then as to not break the game. At least as far as I can tell. 

Side note of this getting more functions than originally planned as I realized that we had essentially made something that just renames the basic array functions.

### CODE REVIEW

File: src/main/java/models/Columns.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        return this.cols.get(columnNumber).get(this.cols.get(columnNumber).size()-1);
+    }
+
+    // remove the top card from the columnFrom column, add it to the columnTo column
+    public void move(int columnFrom, int columnTo) {
+
+        //check if columnFrom is empty, if it's empty, give error; else, run code
+        if (columnHasCards(columnFrom) && !columnHasCards(columnTo)) {
+
+            Card topCard = cols.get(columnFrom).get(cols.get(columnFrom).size()-1);
```
Comment (Manason) [inline src/main/java/models/Columns.java]: Why not use the getTopCard() method here?

File: src/main/java/models/Columns.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            cols.get(columnFrom).remove(cols.get(columnFrom).size()-1);
```
Comment (Manason) [inline src/main/java/models/Columns.java]: Use the remove() method

File: src/main/java/models/Columns.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            cols.get(columnTo).add(topCard);
```
Comment (Manason) [inline src/main/java/models/Columns.java]: use the addCard() method

File: src/main/java/models/Columns.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        } else {
+            System.out.println("Check piles!");
```
Comment (Manason) [inline src/main/java/models/Columns.java]: This error is not very descriptive. It doesn't let me know that I attempted a move or why the move failed.

File: src/main/java/models/Columns.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    // Determines if a column has cards
+    public boolean columnHasCards(int columnNumber) {
+        // check indicated column for number of cards; if no cards return false, otherwise return true
+        return (cols.get(columnNumber).size() > 0);
+    }
+
+    // Gets the top card of a column
+    public Card getTopCard(int columnNumber) {
+        return this.cols.get(columnNumber).get(this.cols.get(columnNumber).size()-1);
```
Comment (Manason) [inline src/main/java/models/Columns.java]: use columnSize()

File: src/main/java/models/Columns.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    // Gets the size of the column columnNumber
+    public int columnSize(int columnNumber) {
+        return cols.get(columnNumber).size();
+    }
+
+    // Determines if a column has cards
+    public boolean columnHasCards(int columnNumber) {
+        // check indicated column for number of cards; if no cards return false, otherwise return true
+        return (cols.get(columnNumber).size() > 0);
```
Comment (Manason) [inline src/main/java/models/Columns.java]: use columnSize()

**Comment by loughrao** [comment]
Implemented better use of methods. No longer just a copy paste from game.java.

**Pull request merged**